contracts/payment/CancellablePullPayment.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity ^0.4.11;
+
+
+import "./PullPayment.sol";
+
+
+/*
+ * @title CancellablePullPayment
+ * 
+ * @dev Inherits from PullPayment and allows the payment sender to cancel
+ * @dev a payment before it's withdrawn by the reciever.
+ * @dev Inherit from this contract and use asyncSend instead of send.
+ */
+
+contract CancellablePullPayment is PullPayment {
+
+  mapping(address => mapping(address => uint )) public paymentsByPayer;
```

Comment:
@ataddei I believe `msg.sender` gets passed through to `asyncSend` when it's called by another contract because my `CancellablePullPaymentMock` inherits from `CancellablePullPayment` and calls `callSend` twice (in the second test) but is able to only cancel one of the payments making me think that a different address is stored in the 2d mapping based on the `msg.sender`.
``` address payer = msg.sender;```
``` uint refund = paymentsByPayer[payer][payee];```